utils/staking-miner/src/dry_run.rs

```diff
@@ -76,24 +112,40 @@ macro_rules! dry_run_cmd_for { ($runtime:ident) => { paste::paste! {
 		signer: Signer,
 	) -> Result<(), Error> {
 		use $crate::[<$runtime _runtime_exports>]::*;
-		let mut ext = crate::create_election_ext::<Runtime, Block>(shared.uri.clone(), config.at, true).await?;
+		let mut ext = crate::create_election_ext::<Runtime, Block>(
+			shared.uri.clone(),
+			config.at,
+			vec!["Staking".to_string(), "System".to_string(), "Balances".to_string()]
+		).await?;
 		force_create_snapshot::<Runtime>(&mut ext)?;
-		measure_snapshot_size::<Runtime>(&mut ext);
-		let (raw_solution, witness) = crate::mine_unchecked::<Runtime>(&mut ext, config.iterations, false)?;
-		log::info!(target: LOG_TARGET, "mined solution with {:?}", &raw_solution.score);
 
+		let (raw_solution, witness) = crate::mine_unchecked::<Runtime>(&mut ext, config.iterations, false)?;
 		let nonce = crate::get_account_info::<Runtime>(client, &signer.account, config.at)
 			.await?
 			.map(|i| i.nonce)
 			.expect("signer account is checked to exist upon startup; it can only die if it \
-				transfers funds out of it, or get slashed. If it does not exist at this point, \
-				it is likely due to a bug, or the signer got slashed. Terminating."
-			);
+			transfers funds out of it, or get slashed. If it does not exist at this point, \
+			it is likely due to a bug, or the signer got slashed. Terminating."
+		);
 		let tip = 0 as Balance;
 		let era = sp_runtime::generic::Era::Immortal;
-		let extrinsic = ext.execute_with(|| create_uxt(raw_solution, witness, signer.clone(), nonce, tip, era));
+		let extrinsic = ext.execute_with(|| create_uxt(raw_solution.clone(), witness, signer.clone(), nonce, tip, era));
 
 		let bytes = sp_core::Bytes(extrinsic.encode().to_vec());
+		print_info::<Runtime>(client, &mut ext, &raw_solution, bytes.clone()).await;
```

Comment:
what would happen if we didn't block on this call? Would it still print correctly? If so it could be  a performance improvement

Reply:
not sure what you mean, elaborate?